askama_parser/src/node.rs

```diff
@@ -15,79 +17,215 @@ use super::{
+    ///   - {{ user.name }}
+    /// {% endfor %}
+    /// ```
     Loop(Loop<'a>),
+    /// A template inheritance declaration.
+    ///
+    /// ```ignore
+    /// {% extends "base.html" %}
+    /// ```
     Extends(&'a str),
+    /// A block definition.
+    ///
+    /// ```ignore
+    /// {% block title %}Index{% endblock %}
+    /// ```
     BlockDef(Ws, &'a str, Vec<Node<'a>>, Ws),
+    /// Include the specified template file inline here.
+    ///
+    /// ```ignore
+    /// {% include "item.html" %}
+    /// ```
     Include(Ws, &'a str),
+    /// Import macros from another template file.
+    ///
+    /// ```ignore
+    /// {% import "macros.html" as scope %}
+    /// ```
     Import(Ws, &'a str, &'a str),
+    /// A macro declaration.
+    ///
+    /// ```ignore
+    /// {% macro heading(arg) %}
+    /// {{arg}}
+    /// -------
+    /// {% endmacro %}
+    /// ```
     Macro(&'a str, Macro<'a>),
+    /// A raw block.
+    ///
+    /// ```ignore
+    /// {% raw %}
+    /// {{ this * is - not + an % expression }}
+    /// {% endraw %}
+    /// ```
+    ///
+    /// The second and fourth tuple elements are the left- and right-side
+    /// white space surrounding the output, and the third is the value
+    /// itself.
     Raw(Ws, &'a str, &'a str, &'a str, Ws),
+    /// The break statement.
     Break(Ws),
+    /// The continue statement.
     Continue(Ws),
 }
 
+/// The Askama equivalent of a Rust pattern, the target of a match or assignment.
 #[derive(Debug, PartialEq)]
 pub enum Target<'a> {
+    /// Bind the value to a name.
     Name(&'a str),
+    /// Destruture a tuple value.
     Tuple(Vec<&'a str>, Vec<Target<'a>>),
+    /// Destructure a struct value.
     Struct(Vec<&'a str>, Vec<(&'a str, Target<'a>)>),
+    /// Match a numeric literal.
     NumLit(&'a str),
+    /// Match a string literal.
     StrLit(&'a str),
+    /// Match a character literal.
     CharLit(&'a str),
+    /// Match a boolean literal.
     BoolLit(&'a str),
+    /// Match against a path.
     Path(Vec<&'a str>),
 }
 
+/// Whitespace preservation or suppression.
 #[derive(Clone, Copy, Debug, PartialEq)]
 pub enum Whitespace {
     Preserve,
     Suppress,
     Minimize,
 }
 
+/// A for loop syntax node.
 #[derive(Debug, PartialEq)]
 pub struct Loop<'a> {
+    /// The whitespace suppression for the start tag.
     pub ws1: Ws,
+    /// The variable of iteration within the loop.
     pub var: Target<'a>,
+    /// The collection to iterate over.
     pub iter: Expr<'a>,
+    /// An optional condition, which if it evaluates to false should skip that iteration.
     pub cond: Option<Expr<'a>>,
+    /// The body of the loop.
     pub body: Vec<Node<'a>>,
+    /// The whitespace suppression for the else tag.
     pub ws2: Ws,
+    /// The else block of the loop, invoked if the collection is empty.
     pub else_block: Vec<Node<'a>>,
+    /// The whitespace suppression for the end tag.
     pub ws3: Ws,
 }
 
+/// A single branch of a match block.
```

Comment:
I'm not very fond of type aliases these days. I think we should make `When` a proper `struct` with fields, and same for `Cond`.

Reply:
Done.